ddtrace/contrib/grpc/patch.py

```diff
@@ -15,12 +17,23 @@
     distributed_tracing_enabled=True,
 ))
 
+
+# Precedence for the service name:
+# 1) DD_GRPC_SERVICE if defined; or
+# 2) For compatibility, the globally set service + "-grpc-client"; or
+# 3) The fall-back "grpc-client"
+if "DD_GRPC_SERVICE" in os.environ:
+    service = os.getenv("DD_GRPC_SERVICE")
+elif config._get_service():
+    service = "{}-{}".format(config._get_service(), constants.GRPC_SERVICE_CLIENT)
+else:
+    service = constants.GRPC_SERVICE_CLIENT
+
+
 # TODO[tbutt]: keeping name for client config unchanged to maintain backwards
 # compatibility but should change in future
```

Comment:
Can we safely remove this now?

Reply:
Hmm. I think you were maybe meaning that the config name itself should be "grpc-client" (as the server is "grpc-server")?

Reply:
(In which case the TODO still applies.)

Reply:
Good thing I made that super clear in my todo 🤦